compiler/rustc_mir_transform/src/coroutine.rs

```diff
@@ -1231,6 +1231,7 @@ fn create_coroutine_drop_shim<'tcx>(
     drop_clean: BasicBlock,
 ) -> Body<'tcx> {
     let mut body = body.clone();
+    let _ = body.coroutine.take();
```

Comment:
A comment here would be good, something like the description on this PR ("Coroutine drop shims are not themselves coroutines, so erase the "coroutine" field from the body so that helper fns like yield_ty and coroutine_kind properly return None for the drop shim.")